lib/util.mli

```diff
@@ -112,6 +112,15 @@ type 'a delayed = unit -> 'a
 
 val delayed_force : 'a delayed -> 'a
 
+(** [try_finalize f x g y] applies the main code [f] to [x] and
+   returns the result after having applied the finalization
+   code [g] to [y]. If the main code raises the exception
+   [exn], the finalization code is executed and [exn] is raised.
+   If the finalization code itself fails, the exception
+   returned is always the one from the finalization code.
+   Credit X.Leroy, D.Remy. *)
+val try_finalize: ('a -> 'b) -> 'a -> ('c -> unit) -> 'c -> 'b
```

Comment:
a big +1, I think I had to write this code many times

minor remark: `finalize -> finally` (as in java, and also as in the name of the variable binding `g` in the implementantion

Reply:
Indeed why not follow the java naming here.